### DISCUSSION TITLE
Getting 404 Not Found for the web interface

### DISCUSSION THREAD

**Discussion by neoblizz** [apache/couchdb#3691] [Q&A]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

(misclicked submitted the issue) I have installed couchdb on amazing aws (lightsail) and have reproduced this on many instances. Basically, my instance tells me that couchdb service is running fine, I can execute the following command with an OK status:

```bash
curl -I http://127.0.0.1:5984/
HTTP/1.1 200 OK
Cache-Control: must-revalidate
Content-Length: 247
Content-Type: application/json
Date: Wed, 04 Aug 2021 02:20:10 GMT
Server: CouchDB/3.1.1 (Erlang OTP/20)
X-Couch-Request-ID: 78da7eef54
X-CouchDB-Body-Time: 0
```

And when I check the `_utils` directory, I get the following using curl:

```bash
curl -I http://127.0.0.1:5984/_utils/
HTTP/1.1 200 OK
Cache-Control: private, must-revalidate
Content-Length: 2009
Content-Security-Policy: child-src 'self' data: blob:; default-src 'self'; img-src 'self' data:; font-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline';
Content-Type: text/html
Date: Wed, 04 Aug 2021 02:20:07 GMT
last-modified: Thu, 17 Sep 2020 21:52:16 GMT
Server: CouchDB/3.1.1 (Erlang OTP/20)
```

The odd thing I have noticed is when the couchdb service is running, I am unable to do anything without getting the following error (even though it shows to be running):

```
 /opt/couchdb/bin/couchdb
Protocol 'inet_tcp': the name couchdb@[my.domain.name.com] seems to be in use by another Erlang node
```
I do not understand the error above because I have tried changing the name (to `couchdb1`) and the domain name to IP or DNS, but no matter what I do, it seems like the error pops up that it is being used. I do not have another couchdb instance on the same node.

Moreover, when I stop the couchdb service, I can use the ` /opt/couchdb/bin/couchdb` binary, but I get the following error:
```
sudo service couchdb stop
/opt/couchdb/bin/couchdb --help
{"Kernel pid terminated",application_controller,"{application_start_failure,config,{{shutdown,{failed_to_start_child,config,{{case_clause,{error,eacces}},[{config,parse_ini_file,1,[{file,\"src/config.erl\"},{line,347}]},{config,'-init/1-fun-0-',1,[{file,\"src/config.erl\"},{line,227}]},{lists,map,2,[{file,\"lists.erl\"},{line,1239}]},{config,init,1,[{file,\"src/config.erl\"},{line,226}]},{gen_server,init_it,2,[{file,\"gen_server.erl\"},{line,365}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,333}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,247}]}]}}},{config_app,start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,config,{{shutdown,{failed_to_start_child,config,{{case_clause,{error,eacces}},[{config,parse_ini_file,1,[{file,"src/config.er

Crash dump is being written to: erl_crash.dump...
```

## Steps to Reproduce

I used Amazon's lightsail free instance, and installed couchdb using apt-get with the steps described here:
https://docs.couchdb.org/en/3.1.1/install/unix.html

## Expected Behaviour

I am expecting a GUI interface available at `my.domain.name/_utils/` (I have nginx routed to `127.0.0.1:5984`)

## Your Environment

Running on Amazon Lightsail (Ubuntu 20.04 --- 5.4.0-1018-aws)
Accessing domain/the GUI using chrome (yes, I have cleared cache)
```
{"couchdb":"Welcome","version":"3.1.1","git_sha":"ce596c65d","uuid":"a86220f2084b969efea38c5f2e4667d0","features":["access-ready","partitioned","pluggable-storage-engines","reshard","scheduler"],"vendor":{"name":"The Apache Software Foundation"}}
```
See the directory structure and permissions below:
```
ll /opt/couchdb/
total 140
drwxr-xr-x  9 root    root      4096 Aug  4 00:57 ./
drwxr-xr-x  3 root    root      4096 Aug  4 00:57 ../
-rw-r--r--  1 root    root    104255 Sep 17  2020 LICENSE
drwxr-xr-x  2 root    root      4096 Aug  4 00:57 bin/
lrwxrwxrwx  1 root    root        16 Sep 17  2020 data -> /var/lib/couchdb/
drwxr-xr-x  8 root    root      4096 Aug  4 00:57 erts-9.3.3.14/
drwxr-xr-x  4 couchdb couchdb   4096 Aug  4 01:36 etc/
drwxr-xr-x 49 root    root      4096 Aug  4 00:57 lib/
drwxr-xr-x  3 root    root      4096 Aug  4 00:57 releases/
drwxr-xr-x  5 root    root      4096 Aug  4 00:57 share/
drwxr-xr-x  2 root    root      4096 Aug  4 00:57 var/
```

## Additional Context
I can also provide the domain name if you'd like to try it yourself (It's not private/personal).

**Comment by neoblizz** [comment]
Pinging this to see if anyone has successfully set up `couchdb` on amazon lightsail? This is my specific use case: https://docs.apexo.app/#server

**Comment by big-r81** [comment]
Hi neoblizz, did you resolve your problem?

**Reply by neoblizz** [reply]
Unfortunately, I did not. I'd be willing to set it all back up if someone can help me with it.

**Comment by big-r81** [comment]
First of all, update CouchDB to version 3.2.2!

Afterwards, tell me what you want to do or what your problems are?

**Reply by neoblizz** [reply]
I'll spend this weekend setting this up again, since last I tried it was a year-ish ago. Will post the problems here, thank you!
